besu/src/main/java/org/hyperledger/besu/cli/options/SynchronizerOptions.java

```diff
@@ -445,7 +457,7 @@ public SynchronizerConfiguration.Builder toDomainObject() {
             .trienodeCountPerRequest(snapsyncTrieNodeCountPerRequest)
             .localFlatAccountCountToHealPerRequest(snapsyncFlatAccountHealedCountPerRequest)
             .localFlatStorageCountToHealPerRequest(snapsyncFlatStorageHealedCountPerRequest)
-            .isSnapServerEnabled(snapsyncServerEnabled)
+            .isSnapServerEnabled(snapsyncServerEnabled || snapsyncServerEnabledExperimental)
```

Comment:
This isn't the flow we typically use when we promote something out of experimental.  Typically we just alias the -X and after a quarter or so just remove the flag.  We can add a deprecation message without adding an additional flag .

That said, this isn't a horrible pattern.  At a minimum there needs to be a CHANGELOG.md entry for this PR, and some text in the upcoming deprecation body that references it.